cloud/scope/powervs_cluster.go

```diff
@@ -1334,16 +1334,26 @@ func (s *PowerVSClusterScope) createTransitGateway() (*string, error) {
 		return nil, fmt.Errorf("error getting resource group id for resource group %v, id is empty", s.ResourceGroup())
 	}
 
-	vpcRegion := s.getVPCRegion()
-	if vpcRegion == nil {
-		return nil, fmt.Errorf("failed to get vpc region")
+	location, globalRouting, err := genUtil.GetTransitGatewayLocationAndRouting(s.Zone(), s.VPC().Region)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get transit gateway location and routing: %w", err)
+	}
+
+	// throw error when user tries to use local routing where global routing is required.
+	// TODO: Add a webhook validation for below condition.
+	if s.IBMPowerVSCluster.Spec.TransitGateway.GlobalRouting != nil && !*s.IBMPowerVSCluster.Spec.TransitGateway.GlobalRouting && *globalRouting {
+		return nil, fmt.Errorf("failed to use local routing for transit gateway since powervs and vpc region used requires global routing")
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to use local routing for transit gateway since powervs and vpc are in different region and requires global routing")
```

Reply:
Fixed it. PTAL now!